### ISSUE TITLE
Add `--all` option to `path`

### ISSUE THREAD

**Issue by IlanCosman** [fish-shell/fish-shell#9921]
`path filter foo bar baz` / `path is foo bar baz` currently return whether _any_ of `foo`, `bar`, or `baz` pass the filtering criteria.

To check whether _all_ the arguments pass the filter, you currently need to use the terrible double-negate:

`not path filter --invert foo bar baz`

which is obviously very unergonomic and not fun to reason about.

I propose adding a `--all` option to `path` to make this useful functionality simpler.

**Comment by gopal-lohar** [comment]
I want to work on this issue

**Comment by dmefs** [comment]
I have a question.
What is the expected result of `path filter --all --invert foo bar baz`
Is it true if none of the arguments pass and false otherwise ?

**Comment by gopal-lohar** [comment]
assumption: the command prints the inputs which passes filter. when using these two options together there are only two possibilities as result, we either print all of them or none of them (since we are using the --all flag it is intuitive that the answer is either all of none)

if i am right about this assumption then let's say printing all is returning true and printing none is returning false

we can apply either of `--all` or `--invert` first

consider this example `path filter --all --invert foo bar baz`

1. apply --invert first then apply --all
    we first list whether each path satisfy the filter or not, then invert each result and check if all results are true or not. so in this example if none of `foo bar baz` exists then it returns true, else return false

2. apply --all first then apply --invert
    we first check if all paths satisfy the filter and then invert the result. so in this example if all of the `foo bar baz` exist then we return false else we return true

in the (2) we will be printing all the paths in many cases when a few of them passes the condition and a few of them don't which is not very intuitive so we should go with (1)


| path                                  |       |       |       |       |       |       |       |       |
| ------------------------------------- | ----- | ----- | ----- | ----- | ----- | ----- | ----- | ----- |
| foo                                   | true  | true  | true  | true  | false | false | false | false |
| bar                                   | true  | true  | false | false | true  | true  | false | false |
| baz                                   | true  | false | true  | false | true  | false | true  | false |
| apply --invert first then apply --all | false | false | false | false | false | false | false | true  |
| apply --all first then apply --invert | false | true  | true  | true  | true  | ture  | ture  | true  |

**Comment by dmefs** [comment]
> in the (2) we will be printing all the paths in many cases when a few of them passes the condition and a few of them don't which is not very intuitive so we should go with (1)

I agree with you. As far as I know, the behavior is the same as `not path filter --invert`. It produces no output, only the exit status is returned.

**Comment by phisonate** [comment]
Shouldn't this be closed per #11507?
